### ISSUE TITLE
Changing CreateTranslator signature

### ISSUE THREAD

**Issue by ahives** [phatboyg/Machete#40]
Currently, CreateTranslator looks like this...

Machete.Schema.CreateTranslator(typeof(MessageTranslate), () => new MessageTranslate());

Perhaps we could refactor so that devs don't have to pass in the type. It could look something like this...

Machete.Schema.CreateTranslator<MessageTranslate>(() => new MessageTranslate());

or it can get the type from the input translate like this...

Machete.Schema.CreateTranslator(() => new MessageTranslate());

**Comment by phatboyg** [comment]
Yeah, I'm thinking we can change this a lot and go to a factory model similar to how Automatonymous does it for activities.

**Comment by ahives** [comment]
Yeah was a little confused at first with the current syntax because CreateTranslator sort of implies that a new Translator is created when in reality the Translator is executing a new Translate. So, instead of CreateTranslate it should be just .Translate(...) similar to .Query(...).

Also, some of other things would be:

1. Drop the word Map from the names of Translate base classes like HL7ComponentTranslateMap, HL7SegmentTranslateMap, etc.

2. Change signature of base classes from:
    public abstract class HL7SegmentTranslateMap<TResult, TInput, TSchema> :
        HL7EntityTranslateMap<TResult, TInput, TSchema>
        where TSchema : HL7Entity
        where TInput : TSchema, HL7Segment
        where TResult : TSchema, HL7Segment

...and...

    public abstract class HL7ComponentTranslateMap<TResult, TInput, TSchema> :
        HL7EntityTranslateMap<TResult, TInput, TSchema>
        where TSchema : HL7Entity
        where TInput : TSchema, HL7Component
        where TResult : TSchema, HL7Component

In all the examples we have, TResult and TInput are the same so why different inputs. For example,
HL7ComponentTranslateMap<MSG, MSG, HL7Entity>

...could look like this...

HL7ComponentTranslate<MSG, HL7Entity>

**Comment by phatboyg** [comment]
Yeah but Map is consistent with other usages, although people using Machete aren't defining their own schema, so they'd be unfamiliar with the `Map` suffix anyway.

In this case, it's really to create a translator, right? So...

`HL7ComponentTranslator<MSG, HL7Entity>`

**Comment by ahives** [comment]
I still like the word Translator to be reserved for the engine and Translate be reserved for describing the action.

**Comment by phatboyg** [comment]
So I think I've converted this all to use the default constructor.
